src/api.cpp

```diff
@@ -5028,11 +5081,24 @@ cl_int CLVK_API_CALL clEnqueueWriteImage(
     if (image->has_any_flag(CL_MEM_HOST_READ_ONLY | CL_MEM_HOST_NO_ACCESS)) {
         return CL_INVALID_OPERATION;
     }
-
-    return cvk_enqueue_image_copy(
-        command_queue, CL_COMMAND_WRITE_IMAGE, image, blocking_write, origin,
-        region, input_row_pitch, input_slice_pitch, const_cast<void*>(ptr),
-        num_events_in_wait_list, event_wait_list, event);
+    auto internal_image = static_cast<cvk_image*>(image);
+    if (internal_image->is_backed_by_buffer_view()) {
+        auto cmd = new cvk_command_buffer_host_copy(
+            command_queue, CL_COMMAND_WRITE_BUFFER,
+            static_cast<cvk_buffer*>(internal_image->buffer()), ptr,
+            origin[0] * internal_image->element_size(),
+            region[0] * internal_image->element_size());
+        auto err = command_queue->enqueue_command_with_deps(
+            cmd, blocking_write, num_events_in_wait_list, event_wait_list,
+            event);
+        return err;
```

Comment:
I think you probably want to push this code to `cvk_enqueue_image_copy` so it's the same for read image and write image commands.

Reply:
Done